src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1090,6 +1090,45 @@ VideoDecoder::BatchDecodedOutput VideoDecoder::getFramesAtIndices(
   return output;
 }
 
+VideoDecoder::BatchDecodedOutput VideoDecoder::getFramesDisplayedByTimestamps(
+    int streamIndex,
+    const std::vector<double>& framePtss) {
+  validateUserProvidedStreamIndex(streamIndex);
+  validateScannedAllStreams("getFramesDisplayedByTimestamps");
+
+  // The frame displayed at timestamp t and the one displayed at timestamp `t +
+  // eps` are probably the same frame, with the same index. The easiest way to
+  // avoid decoding that unique frame twice is to convert the input timestamps
+  // to indices, and leverage the de-duplication logic of getFramesAtIndices.
+
+  const auto& streamMetadata = containerMetadata_.streams[streamIndex];
+  const auto& stream = streams_[streamIndex];
+  double minSeconds = streamMetadata.minPtsSecondsFromScan.value();
+  double maxSeconds = streamMetadata.maxPtsSecondsFromScan.value();
+
+  std::vector<int64_t> frameIndices(framePtss.size());
+  for (auto i = 0; i < framePtss.size(); ++i) {
+    auto framePts = framePtss[i];
+    TORCH_CHECK(
+        framePts >= minSeconds && framePts < maxSeconds,
+        "frame pts is " + std::to_string(framePts) + "; must be in range [" +
+            std::to_string(minSeconds) + ", " + std::to_string(maxSeconds) +
+            ").");
+
+    auto it = std::lower_bound(
+        stream.allFrames.begin(),
+        stream.allFrames.end() - 1,
```

Comment:
Are you intentionally not including the last frame? Note that by convention, the `.end()` method on C++ containers is actually an iterator _past_ the end: https://en.cppreference.com/w/cpp/container/vector/end. So this code is actually eliminating the final element from the range.

Reply:
Hmmm. I think you're right. We're eliminating the final element, but if we should have gotten the final element, it will become that anyway. This works for, I think, two resons:

1. Because we've already eliminated the possibility of a pts value outside of the valid range because of the `TORCH_CHECK` above.
2. Because we're converting iterators to indices, and `.end()` becomes equivalent to `.size()` with the subtraction method we're using.

But, I think I actually prefer using `min` to fix things up after. That this works is so subtle, and doing `.end() - 1` in an iterator range is so unusual, I'd rather have the `min` approach with a comment explaining why we need it. I think it makes the code more understandable. The comment would be something like:
```
// If the frame index is larger than the size of all frames, that means we 
// couldn't match the pts value to the pts value of a NEXT FRAME. And 
// that means that this timestamp falls during the time between when the 
// last frame is displayed, and the video ends. Hence, it should map to the
// index of the last frame.